code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -34,6 +34,32 @@
 		M.hallucination += 5
 	..()
 
+/datum/reagent/drug/cannabis
+	name = "Cannabis"
+	description = "A psychoactive drug from the Cannabis plant used for recreational purposes."
+	color = "#059033"
+	overdose_threshold = INFINITY
+	ph = 6
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	metabolization_rate = 0.125 * REAGENTS_METABOLISM
+
+/datum/reagent/drug/cannabis/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	M.apply_status_effect(/datum/status_effect/stoned)
```

Comment:
This shouldn't be re-applied on every mob life cycle.
Make it a status effect without maximum duration and apply it on first metabolization, removing it when the chem leaves the body.
Make sure the metabolization rate is slow.

Reply:
It took me so many hours to get the red eyes to work. I didn't use a status effect originally, but oranges told me a timed status effect would get it to work and it did. I tried different metabolisation rates (such as the one used by meth, which applies a movement speed modifier), but anytime the reagent added the red eyes they would immediately disappear again. I'm guessing the same thing will happen with the status effect. I'll try it, though. I don't know a lot about how reagent metabolisation works.

Reply:
The status effect is fine. Just add it on first metabolisation and remove it on last, and make sure the status effect does not have a set duration, but that it stays there until removed.

Reply:
I'll try it but what I'm saying is I don't think it will work if the status effect isn't timed. It will just disappear immediately, like with the red eyes trait.

Reply:
Yeah it doesn't work

Reply:
There's status effects that don't disappear based on time.
Check for examples in other effects, and the status effects variables and their documentation.

Reply:
I know how to add a status effect that doesn't disappear based on time. 

The issue is as I've described above. 

I tried for hours to get this to work using on_mob_metabolize() and on_mob_end_metabolize() (the way reagents like methamphetamine add and remove their movespeed modifiers) before oranges suggested using a timed status effect. Then it finally worked, which was a huge relief.

I just tried using on_mob_metabolize() and on_mob_end_metabolize() again with the status effect. I set the status effect to not have a duration. It appeared to not be applied at all.

When I was trying to get it to work before the status effect, I used debug messages to discover that the things being applied with on_mob_metabolize() were immediately being removed by on_mob_end_metabolize().

No amount of changing the metabolisation rate helped. I tried using methamphetamine's metabolisation rate, because it applies a movement speed modifier that sticks. No luck. I would've tried other metabolisation rates too.



You must know from personal experience how tedious it is to compile the code, wait for the world to finish setting up and get into testing it.

Reply:
I know how to add a status effect that doesn't disappear based on time. 

The issue is as I've described above. 

I tried for hours to get this to work using on_mob_metabolize() and on_mob_end_metabolize() (the way reagents like methamphetamine add and remove their movespeed modifiers) before oranges suggested using a timed status effect. Then it finally worked, which was a huge relief.

I just tried using on_mob_metabolize() and on_mob_end_metabolize() again with the status effect. I set the status effect to not have a duration. It appeared to not be applied at all.

When I was trying to get it to work before the status effect, I used debug messages to discover that the things being applied with on_mob_metabolize() were immediately being removed by on_mob_end_metabolize().

No amount of changing the metabolisation rate helped. I tried using methamphetamine's metabolisation rate, because it applies a movement speed modifier that sticks. No luck. I would've tried other metabolisation rates too.



You must know from personal experience how tedious it is to compile the code, wait for the world to finish setting up and get into testing it.

Reply:
I tried for hours to get it to work with on_mob_metabolise() and on_mob_end_metabolise() the way you're asking for now. It was incredibly frustrating. @optimumtact suggested using a time status effect and it worked. I have no idea how to get it to work the way you're asking for and I don't want to go through trying to get it to work that way again.

Reply:
@Rohesie the way cigs microdose reagents mean it's constantly added and removed

Reply:
I see. If that's the case this will be hacky one way or another, until a better system is cooked.